### ISSUE TITLE
Kaminari support

### ISSUE THREAD

**Issue by aaronchi** [karmi/retire#14]
It would be nice if you guys added Kaminari support to the pagination module. Kaminari seems to be replacing will_paginate as the preferred pagination solution for Rails.

I haven't looked at in in detail but I believe this is pretty easy and would just involve aliasing some methods on the pagination module.

**Comment by khoan** [comment]
monkey patch

``` ruby
module Tire
  module Results
    module Pagination
      alias :num_pages :total_pages

      def limit_value
        @options[:per_page] ? @options[:per_page].to_i : 10
      end

    end
  end
end
```

**Comment by karmi** [comment]
Sorry, I haven't any time to look into this lately -- also I never heard about _kaminari_ :)

I think the compability should be easy to add... thanks @khoan for the pointer.

**Comment by khoan** [comment]
glad to be of service.

for a quick 'n' lite intro to kaminari http://railscasts.com/episodes/254-pagination-with-kaminari

**Comment by kbaum** [comment]
Hi.  I tried the monkey patch and I tried to follow the screencast but i get a missing method page.

```
undefined method `page' for #<Tire::Results::Collection:0x0000010547ef18>
```

Which makes sense because we don't have a page method from the patch.  From my controller i have:

  @email_messages = EmailMessage.search(params[:simple_search][:value]).page(params[:page])

When I get rid of the page method, then i run into a division by 0 error which probably has something to do with the page being 0?

```
Rendered simple_searches/index.html.erb within layouts/application (1.4ms)
```

Completed 500 Internal Server Error in 353ms

```
ActionView::Template::Error (divided by 0):
4: <% end %>
5: 
6: <% if @email_messages.present? %>
7:     <%= paginate @email_messages %>
8:     <div id="products">
9:       <%= render @email_messages %>
10:     </div>
  app/views/simple_searches/index.html.erb:7:in `_app_views_simple_searches_index_html_erb__815704194711493677_2177278500__808843748140322580
```

**Comment by kbaum** [comment]
So you just have to pass the page in as an options hash

```
 @email_messages = EmailMessage.search(params[:simple_search][:value], :page=>1)
```

**Comment by kbaum** [comment]
Hmmm.. only problem with this approach is that i am always on page 1.  Even if i change my code to this:

```
@email_messages = EmailMessage.search(params[:simple_search][:value], :page=>(params[:page] || 1) )
```

**Comment by kbaum** [comment]
Ok, i couldn't get this working so i went back to will_paginate which i now have working.

**Comment by karmi** [comment]
I'll definitely look into supporting _kaminari_ as soon as I clean percolater and other stuff piled up.

**Comment by ramontayag** [comment]
Yeah this will be cool. I'll use will_paginate for now :)

**Comment by karmi** [comment]
Closed by karmi/tire#105.
